Refresh network-digi-sarian-snmp (#3676)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/network-digi-sarian-snmp.md

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash
/usr/lib/centreon/plugins/centreon_digi_sarian_snmp.pl \
	--plugin=network::digi::sarian::snmp::plugin \
	--mode=interfaces \
	--hostname=10.0.0.1 \
	--snmp-version='2c' \
	--snmp-community='my-snmp-community'  \
	--interface='.*' \
	--name \
	--add-status \
	--add-traffic \
	--critical-status='' \
	--warning-in-traffic='80' \
	--critical-in-traffic='90' \
	--warning-out-traffic='80' \
	--critical-out-traffic='90' \
	--verbose
```

The expected command output is shown below:

```bash

> OK: All interfaces are ok | 'interface_name1#interface.traffic.in.bitspersecond'=48034b/s;;80;90; 'interface_name2#interface.traffic.in.bitspersecond'=79181b/s;;80;90; 'interface_name1#interface.traffic.out.bitspersecond'=60438b/s;;80;90; 'interface_name2#interface.traffic.out.bitspersecond'=1708b/s;;80;90; 

Author: ```suggestion
OK: All interfaces are ok | 'interface_name1#interface.traffic.in.bitspersecond'=48034b/s;80;90;; 'interface_name2#interface.traffic.in.bitspersecond'=79181b/s;80;90;;  'interface_name1#interface.traffic.out.bitspersecond'=60438b/s;80;90;;  'interface_name2#interface.traffic.out.bitspersecond'=1708b/s;80;90;;  
```

> OK: All interfaces are ok | 'interface_name1#interface.traffic.in.bitspersecond'=48034b/s;80;90;; 'interface_name2#interface.traffic.in.bitspersecond'=79181b/s;80;90;;  'interface_name1#interface.traffic.out.bitspersecond'=60438b/s;80;90;;  'interface_name2#interface.traffic.out.bitspersecond'=1708b/s;80;90;;  

Author: ```suggestion
OK: All interfaces are ok | 'interface_name1#interface.traffic.in.bitspersecond'=20b/s;80;90;; 'interface_name2#interface.traffic.in.bitspersecond'=20b/s;80;90;;  'interface_name1#interface.traffic.out.bitspersecond'=20b/s;80;90;;  'interface_name2#interface.traffic.out.bitspersecond'=20b/s;80;90;;  
```

| status     | N/A   |
| technology | N/A   |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
<TabItem value="Memory" label="Memory">

| Name | Unit  |
|:-----|:------|
| used | B     |

</TabItem>
<TabItem value="Temperature" label="Temperature">

| Name      | Unit  |
|:----------|:------|
| device    | C     |
| processor | C     |
| modem     | C     |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
<TabItem value="Traffic-Global" label="Traffic-Global">

| Name                                                  | Unit  |
|:------------------------------------------------------|:------|
| status                                                | N/A   |
| status                                                | N/A   |
| *interface_name1*#interface.traffic.in.bitspersecond  | b/s   |
| *interface_name2*#interface.traffic.in.bitspersecond  | b/s   |
| *interface_name1*#interface.traffic.out.bitspersecond | b/s   |
| *interface_name2*#interface.traffic.out.bitspersecond | b/s   |

</TabItem>
</Tabs>

## Prerequisites

### SNMP Configuration

> To use this pack, the SNMP service must be properly configured on your **Digi Sarian** equipment.

Author: ```suggestion
The SNMP agent must be configured and activated on the host. Please refer to the official documentation from the manufacturer/publisher.
```

### Network flow

> The target resource must be reachable from the Centreon poller on the UDP/161
> SNMP port.

Author: ```suggestion
The target resource must be reachable from the Centreon poller on the UDP/161 SNMP port.
```

## Installing the monitoring connector

### Pack

1. If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the connector displayed within the
**Configuration > Monitoring Connector Manager** menu.
If the platform uses an *offline* license, install the package on the **central server**
with the command corresponding to the operating system's package manager:

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-pack-network-digi-sarian-snmp
```

</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf install centreon-pack-network-digi-sarian-snmp
```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

```bash
apt install centreon-pack-network-digi-sarian-snmp
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-pack-network-digi-sarian-snmp
```

</TabItem>
</Tabs>

> 2. Whatever the license type (*online* or *offline*), install the **Digi Sarian** connector through

Reviewer: ```suggestion
2. Whatever the license type (*online* or *offline*), install the **Digi Sarian SNMP** connector through
```

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash
/usr/lib/centreon/plugins/centreon_digi_sarian_snmp.pl \
	--plugin=network::digi::sarian::snmp::plugin \
	--mode=interfaces \
	--hostname=10.0.0.1 \
	--snmp-version='2c' \
	--snmp-community='my-snmp-community'  \
	--interface='.*' \
	--name \
	--add-status \
	--add-traffic \
	--critical-status='' \
	--warning-in-traffic='80' \
	--critical-in-traffic='90' \

> 	--warning-out-traffic='80' \

Reviewer: pourcentage?

	--critical-out-traffic='90' \
	--verbose
```

The expected command output is shown below:

```bash

> OK: All interfaces are ok | 'interface_name1#interface.traffic.in.bitspersecond'=48034b/s;;80;90; 'interface_name2#interface.traffic.in.bitspersecond'=79181b/s;;80;90; 'interface_name1#interface.traffic.out.bitspersecond'=60438b/s;;80;90; 'interface_name2#interface.traffic.out.bitspersecond'=1708b/s;;80;90; 

Reviewer: place des valeurs?
